### PR TITLE
Changes related to list data centre api

### PR DESCRIPTION
**Pull request by Azher2Ali**
Adding a new REST endpoint for retrieving a list of all Data Centers. Each Data Center will include connection details. This endpoint is publicly accessible, and Data Centers will be returned as a JSON array, excluding their associated programs. A separate endpoint will be created to retrieve programs within a specific Data Center.

### CODE REVIEW

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java [hunk 1]
```diff
@@ -319,4 +319,12 @@ public ResponseEntity<AddInstitutionsResponseDTO> addInstitutions(
...
   }
+
+  @GetMapping(value = "/dataCenter")
+  public ResponseEntity<List<DataCenterDTO>> listDataCenters(
+      @Parameter(hidden = true) @RequestHeader(value = "Authorization", required = true)
+          final String authorization) {
+    val dataCenterEntities = serviceFacade.listDataCenters();
+    return new ResponseEntity(dataCenterEntities, HttpStatus.OK);
+  }
 }
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/controller/ProgramController.java:330]: Should we have a new controller for data centers?

File: src/main/java/org/icgc/argo/program_service/converter/ProgramConverter.java [hunk 2]
```diff
@@ -57,6 +58,8 @@ public interface ProgramConverter {
   @Mapping(target = "dataCenterId", ignore = true)
   ProgramEntity programToProgramEntity(Program p);
 
+  DataCenterDTO dataCenterToDataCenterEntity(DataCenterEntity p);
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/converter/ProgramConverter.java:61]: Similarly, a new converter for Data Centers, this is a separate entity which should keep its own code separate

**Review by joneubank — CHANGES_REQUESTED** [review]
Looks good, just needs a bit more code organization.

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java (hunk 1 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/controller/ProgramController.java]: Moved the endpoint to DataCenterController

File: src/main/java/org/icgc/argo/program_service/converter/ProgramConverter.java (hunk 2 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/converter/ProgramConverter.java]: Moved the mapping to DataCenterConvertor

**Review by joneubank — APPROVED** [review]

**Pull request merged**